draft-ietf-tls-md5-sha1-deprecate-07.xml

```diff
@@ -172,10 +172,10 @@
 
    <section anchor="Signature_algorithms" title="Signature Algorithms">
    <t>
-   Clients MUST NOT include MD5 and SHA-1 in the signature_algorithms extension. If
-   a client does not send a signature_algorithms extension, then the server
-   MUST abort the handshake and send a handshake_failure alert, except when
-   digital signatures are not used (for example, when using PSK ciphers).
+   Clients MUST include the signature_algorithms extension. Clients MUST NOT include MD5
+   and SHA-1 in it this extension. If a client does not send a signature_algorithms
+   extension, then the server MUST abort the handshake and send a handshake_failure alert,
+   except when digital signatures are not used (for example, when using PSK ciphers).
```

Comment:
"MUST abort ... except when .... are not used (for example ...)" is very hard to parse. :-)

There are also non-PSK examples like RSA cipher suites which... either do or don't look at signature_algorithms depending on how much you believe the server is supposed to check its own certificate chain. What if we moved this to the ServerKeyExchange bit and say:

> Servers MUST NOT include MD5 and SHA-1 in ServerKeyExchange messages. If no other signature algorithms are available (for example, if the client does not send a signature_algorithms extension), the server MUST abort the handshake with a handshake_failure alert or select a different cipher suite.

Reply:
Simpler -- I'll take it!